Build failure: libvirt (darwin)

### Steps To Reproduce

Steps to reproduce the behavior:
1. build *X*

### Build log

https://hydra.nixos.org/job/nixpkgs/staging-next/libvirt.aarch64-darwin

```
clang -Isrc/util/libvirt_util.a.p -Isrc/util -I../src/util -Iinclude -I../include -Isrc -I../src -I. -I.. -I/nix/store/48vq8kbm919swql77jxr1b4iq3lcf2ff-gnutls-3.8.6-dev/include -I/nix/store/qqykkpgl0f02cyx8c2pzssb3mn8mvp1g-glib-2.82.1-dev/include/glib-2.0 -I/nix/store/6a4qvifcnrd42jd6fb8vrxm4a5d7rp2d-glib-2.82.1/lib/glib-2.0/include -I/nix/store/qqykkpgl0f02cyx8c2pzssb3mn8mvp1g-glib-2.82.1-dev/include -I/nix/store/qqykkpgl0f02cyx8c2pzssb3mn8mvp1g-glib-2.82.1-dev/include/gio-unix-2.0 -I/nix/store/j5jncsgnbi91i46znqagnphh15ncyqy2-libxml2-2.13.4-dev/include/libxml2 -I/nix/store/sn1sbyr6a5j2k5hxs8vmc649wk1w5w9x-yajl-2.1.0-unstable-2024-02-01/include -fdiagnostics-color=always -Wall -Winvalid-pch -Wextra -std=gnu99 -fasynchronous-unwind-tables -fexceptions -fno-common -Wabsolute-value -Waddress -Waddress-of-packed-member -Walloca -Wattribute-warning -Wattributes -Wbool-operation -Wbuiltin-macro-redefined -Wcast-align -Wno-cast-function-type -Wchar-subscripts -Wcomment -Wcomments -Wcpp -Wdangling-else -Wdate-time -Wdeclaration-after-statement -Wdeprecated-declarations -Wdiv-by-zero -Wduplicate-decl-specifier -Wempty-body -Wendif-labels -Wexpansion-to-defined -Wformat-extra-args -Wno-format-nonliteral -Wformat-security -Wformat-y2k -Wformat-zero-length -Wframe-address -Wframe-larger-than=2048 -Wfree-nonheap-object -Wignored-attributes -Wignored-qualifiers -Wimplicit -Wimplicit-function-declaration -Wimplicit-int -Wincompatible-pointer-types -Winit-self -Winline -Wint-conversion -Wint-in-bool-context -Wint-to-pointer-cast -Winvalid-pch -Wlogical-not-parentheses -Wmain -Wmemset-transposed-args -Wmisleading-indentation -Wmissing-braces -Wmissing-declarations -Wmissing-field-initializers -Wmissing-include-dirs -Wmissing-prototypes -Wmultichar -Wnarrowing -Wnested-externs -Wnonnull -Wnull-dereference -Wodr -Wold-style-definition -Woverflow -Woverride-init -Wparentheses -Wpointer-arith -Wpointer-compare -Wpointer-sign -Wpointer-to-int-cast -Wpragmas -Wpsabi -Wreturn-local-addr -Wreturn-type -Wsequence-point -Wshadow -Wshift-count-negative -Wshift-count-overflow -Wshift-negative-value -Wno-sign-compare -Wsizeof-array-argument -Wsizeof-pointer-div -Wsizeof-pointer-memaccess -Wstrict-aliasing -Wstrict-prototypes -Wswitch -Wswitch-bool -Wswitch-enum -Wtautological-compare -Wtrigraphs -Wtype-limits -Wno-typedef-redefinition -Wuninitialized -Wunknown-pragmas -Wunused -Wunused-but-set-parameter -Wunused-but-set-variable -Wunused-function -Wunused-label -Wunused-local-typedefs -Wunused-parameter -Wunused-result -Wunused-value -Wunused-variable -Wvarargs -Wvariadic-macros -Wvla -Wvolatile-register-var -Wwrite-strings -fstack-protector-strong -Wdouble-promotion -Wno-unused-function -DIN_LIBVIRT '-Dabs_top_builddir="/private/tmp/nix-build-libvirt-10.5.0.drv-0/source/build"' '-Dabs_top_srcdir="/private/tmp/nix-build-libvirt-10.5.0.drv-0/source"' -MD -MQ src/util/libvirt_util.a.p/vireventthread.c.o -MF src/util/libvirt_util.a.p/vireventthread.c.o.d -o src/util/libvirt_util.a.p/vireventthread.c.o -c ../src/util/vireventthread.c
../src/util/vireventthread.c:35:49: error: too many arguments provided to function-like macro invocation
G_DEFINE_TYPE(virEventThread, vir_event_thread, G_TYPE_OBJECT)
                                                ^
/nix/store/qqykkpgl0f02cyx8c2pzssb3mn8mvp1g-glib-2.82.1-dev/include/glib-2.0/gobject/gtype.h:189:25: note: expanded from macro 'G_TYPE_OBJECT'
#define G_TYPE_OBJECT                   G_TYPE_MAKE_FUNDAMENTAL (20)
                                        ^
/nix/store/qqykkpgl0f02cyx8c2pzssb3mn8mvp1g-glib-2.82.1-dev/include/glib-2.0/gobject/gtype.h:227:36: note: expanded from macro 'G_TYPE_MAKE_FUNDAMENTAL'
#define G_TYPE_MAKE_FUNDAMENTAL(x)      ((GType) ((x) << G_TYPE_FUNDAMENTAL_SHIFT))
                                        ^
/nix/store/qqykkpgl0f02cyx8c2pzssb3mn8mvp1g-glib-2.82.1-dev/include/glib-2.0/gobject/gtype.h:2281:9: note: macro '_G_DEFINE_TYPE_EXTENDED_BEGIN_PRE' defined here
#define _G_DEFINE_TYPE_EXTENDED_BEGIN_PRE(TypeName, type_name) \
        ^
../src/util/vireventthread.c:35:1: error: unknown type name '_G_DEFINE_TYPE_EXTENDED_BEGIN_PRE'
G_DEFINE_TYPE(virEventThread, vir_event_thread, G_TYPE_OBJECT)
```

### Additional context

This broke in the last staging cycle. I also noticed that `glib` was upgraded from 2.80.4 to 2.82.1 in the same cycle.

### Notify maintainers

@fpletz @globin @lovesegfault 

### Metadata

<!-- Please insert the output of running `nix-shell -p nix-info --run "nix-info -m"` below this line -->

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/applications/virtualization/virt-manager/default.nix
+++ b/pkgs/applications/virtualization/virt-manager/default.nix
@@ -97,6 +97,7 @@ python3.pkgs.buildPythonApplication rec {
     "test_misc_nonpredicatble_generate"
     "test_disk_dir_searchable"  # does something strange with permissions
     "testCLI0001virt_install_many_devices"  # expects /var to exist
+    "testCLI0263virt_xml"  # depends on a specific libvirt version
   ];
 
   preCheck = ''

--- a/pkgs/by-name/li/libvirt-glib/package.nix
+++ b/pkgs/by-name/li/libvirt-glib/package.nix
@@ -1,6 +1,7 @@
 { lib
 , stdenv
 , fetchurl
+, fetchpatch
 , meson
 , ninja
 , pkg-config
@@ -27,6 +28,14 @@ stdenv.mkDerivation rec {
     url = "https://libvirt.org/sources/glib/${pname}-${version}.tar.xz";
     sha256 = "m/7DRjgkFqNXXYcpm8ZBsqRkqlGf2bEofjGKpDovO4s=";
   };
+
+  patches = [
+    (fetchpatch {
+      name = "relax-max-stack-size-limit.patch";
+      url = "https://gitlab.com/libvirt/libvirt-glib/-/commit/062f21ccaa810087637ae24e0eb69f1a0f0a45f5.patch";
+      hash = "sha256-6mvINDd1HYS7oZsyNiyEwdNJfK5I5nPx86TRMq2RevA=";
+    })
+  ];
 
   nativeBuildInputs = [
     meson

--- a/pkgs/development/libraries/libvirt/default.nix
+++ b/pkgs/development/libraries/libvirt/default.nix
@@ -34,7 +34,7 @@
 , stdenv
 , substituteAll
 , xhtml1
-, yajl
+, json_c
 , writeScript
 , nixosTests
 
@@ -114,13 +114,13 @@ stdenv.mkDerivation rec {
   # NOTE: You must also bump:
   # <nixpkgs/pkgs/development/python-modules/libvirt/default.nix>
   # SysVirt in <nixpkgs/pkgs/top-level/perl-packages.nix>
-  version = "10.5.0";
+  version = "10.9.0";
 
   src = fetchFromGitLab {
     owner = pname;
     repo = pname;
     rev = "v${version}";
-    hash = "sha256-Nku4l1f34NOUr23KWDH9uZu72OgMK3KfYjsRRbuTvf8=";
+    hash = "sha256-LYQYA5UIKYs+8rSNZDymmrxuTWsgmukP5Y17lGB5UQs=";
     fetchSubmodules = true;
   };
 
@@ -202,7 +202,7 @@ stdenv.mkDerivation rec {
     python3
     readline
     xhtml1
-    yajl
+    json_c
   ] ++ lib.optionals isLinux [
     acl
     attr
@@ -312,7 +312,7 @@ stdenv.mkDerivation rec {
       (feat "ssh_proxy" isLinux)
       (feat "tests" true)
       (feat "udev" isLinux)
-      (feat "yajl" true)
+      (feat "json_c" true)
 
       (driver "ch" isLinux)
       (driver "esx" true)

--- a/pkgs/development/libraries/libvirt/default.nix
+++ b/pkgs/development/libraries/libvirt/default.nix
@@ -163,6 +163,9 @@ stdenv.mkDerivation rec {
     sed -i '/qemuvhostusertest/d' tests/meson.build
     sed -i '/qemuxml2xmltest/d' tests/meson.build
     sed -i '/domaincapstest/d' tests/meson.build
+    # virshtest frequently times out on Darwin
+    substituteInPlace tests/meson.build \
+      --replace-fail "data.get('timeout', 30)" "data.get('timeout', 120)"
   '' + lib.optionalString enableXen ''
     # Has various hardcoded paths that don't exist outside of a Xen dom0.
     sed -i '/libxlxml2domconfigtest/d' tests/meson.build

--- a/pkgs/development/python-modules/libvirt/default.nix
+++ b/pkgs/development/python-modules/libvirt/default.nix
@@ -11,14 +11,14 @@
 
 buildPythonPackage rec {
   pname = "libvirt";
-  version = "10.5.0";
+  version = "10.9.0";
   pyproject = true;
 
   src = fetchFromGitLab {
     owner = "libvirt";
     repo = "libvirt-python";
     rev = "v${version}";
-    hash = "sha256-dPjT9PRoUzNrY79yejAW/sbkMr0fpLif7IKZIW/K3KI=";
+    hash = "sha256-/kjpB19X90btIewW+hjLjLagJvI5X2oIHXpcSZVtu2I=";
   };
 
   build-system = [ setuptools ];

--- a/pkgs/top-level/perl-packages.nix
+++ b/pkgs/top-level/perl-packages.nix
@@ -24077,12 +24077,12 @@ with self; {
 
   SysVirt = buildPerlModule rec {
     pname = "Sys-Virt";
-    version = "10.2.0";
+    version = "10.9.0";
     src = fetchFromGitLab {
       owner = "libvirt";
       repo = "libvirt-perl";
       rev = "v${version}";
-      hash = "sha256-xpgZeXk9QefqbBMsvcMh/Cg/XFGEiVi3FbU/jBbSIr0=";
+      hash = "sha256-g2HH9Ep5cAa4qXo9/MKJmxeive6oqHQEX9C8qY+u2g4=";
     };
     nativeBuildInputs = [ pkgs.pkg-config ];
     buildInputs = [ pkgs.libvirt CPANChanges TestPod TestPodCoverage XMLXPath ];

